lib/tasks/reception_generator.rb

```diff
@@ -62,7 +62,7 @@ def tubes
     @number_of_tubes.times.map do
       barcode = @barcodes.next
       library_type = @library_types.next
-      data_type = @data_types.next
+      data_type = begin; @data_types.next; rescue StopIteration; nil; end
```

Comment:
This feels a bit clunky to me.  A clever Ruby trick, but (I think) the block below is applied to the result of `.next` in the middle of this line and that's not clear enough IMHO.  Especially with the use of `;` as statement separators.

I would probably prefer to see the error being handled more explicitly across multiple lines.